robot/web/stream/camera/camera.go

```diff
@@ -24,14 +63,116 @@ func Camera(robot robot.Robot, stream gostream.Stream) (camera.Camera, error) {
 	return cam, nil
 }
 
+// FormatStringToMimeType takes a format string returned from image.DecodeConfig and converts
+// it to a utils mime type.
+func FormatStringToMimeType(format string) string {
+	return fmt.Sprintf("image/%s", format)
+}
+
+// GetStreamableNamedImageFromCamera returns the first named image it finds from the camera that is supported for streaming.
+func GetStreamableNamedImageFromCamera(ctx context.Context, cam camera.Camera) (camera.NamedImage, error) {
+	namedImages, _, err := cam.Images(ctx, nil, nil)
+	if err != nil {
+		return camera.NamedImage{}, err
+	}
+	if len(namedImages) == 0 {
+		return camera.NamedImage{}, fmt.Errorf("no images received for camera %q", cam.Name())
+	}
+
+	for _, namedImage := range namedImages {
+		if _, ok := StreamableImageMIMETypes[namedImage.MimeType()]; ok {
+			return namedImage, nil
+		}
+
+		imgBytes, err := namedImage.Bytes(ctx)
+		if err != nil {
+			continue
+		}
+
+		_, format, err := image.DecodeConfig(bytes.NewReader(imgBytes))
+		if err != nil {
+			continue
+		}
+		if _, ok := StreamableImageMIMETypes[FormatStringToMimeType(format)]; ok {
+			return namedImage, nil
```

Comment:
almost all code except the highlighted is directly copy and pasted from `camera2/camera.go` (now deleted since it used to be gated behind a feature flag). The `FormatStringToMimeType` helper is also new along with the corresponding test.

I added this new code because I want the stream server to be more robust i.e. if a mimeType is missing/wrong somehow, we still want to be able to stream if the underlying bytes are healthy. `DecodeConfig` is a lightweight function that checks the header bytes of the actual image bytes and tells us what the format is.